### PR TITLE
102 get donors route

### LINKED ISSUE #102: GET /api/admin/donors
- [x] create route controller and get donors' data from database 
- [x] test route

### PR DESCRIPTION
**Pull request by farah2003**
relates #102 
close #102

### CODE REVIEW

**Review by Mohammed-Hamada — APPROVED** [review]

**Review by DeenaAlghazali — APPROVED** [review]

File: server/controllers/admin/donors/getDonors.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import { NextFunction, Request, Response } from 'express';
+import {
+  Donation, Donor, sequelize,
+} from '../../../database/models';
+import { CustomError } from '../../../utils';
+import { donorsQuerySchema } from '../../../utils/validation';
+
+const getDonors = async (req:Request, res:Response, next:NextFunction) => {
```
Comment (HaniOlwan) [inline server/controllers/admin/donors/getDonors.ts:8]: change the name to getDonorsForCampaign just to be more clear

Comment (HaniOlwan) [inline server/controllers/admin/donors/getDonors.ts:8]: change the name to getDonorsForCampaign just to be more clear

File: server/controllers/admin/donors/getDonors.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  try {
+    const { page = 1, limit = 6 } = req.query;
+    await donorsQuerySchema.validateAsync(req.query);
```
Comment (HaniOlwan) [inline server/controllers/admin/donors/getDonors.ts:11]: short the two lines for 
const { page = 1, limit = 6 } = donorsQuerySchema.validateAsync(req.query);

Comment (HaniOlwan) [inline server/controllers/admin/donors/getDonors.ts:11]: short the two lines for 
const { page = 1, limit = 6 } = donorsQuerySchema.validateAsync(req.query);

**Review by HaniOlwan — APPROVED** [review]

File: server/controllers/admin/donors/getDonors.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
+import { NextFunction, Request, Response } from 'express';
+import {
+  Donation, Donor, sequelize,
+} from '../../../database/models';
+import { CustomError } from '../../../utils';
+import { donorsQuerySchema } from '../../../utils/validation';
```
Comment (Raghad-Mezied) [inline server/controllers/admin/donors/getDonors.ts:6]: Import them in one line from the utils

File: server/database/fakeData/buildFakeData.ts [hunk 4]
```diff
@@ -45,12 +41,16 @@ const buildFakeData = async () => {
     donors.map(async (donor: any) => {
       await Donor.create(donor);
     }),
-    donations.map(async (donation: any) => {
-      await Donation.create(donation);
-    }),
+
+  ]);
+  await Promise.all(campaigns.map((campaign: any) => Campaign.create(campaign)));
```
Comment (Raghad-Mezied) [inline server/database/fakeData/buildFakeData.ts:46]: Move this to the first Promis

**Review by Raghad-Mezied — CHANGES_REQUESTED** [review]

**Review by Raghad-Mezied — APPROVED** [review]

**Pull request merged**